src/test/suite/language/mongoDBService.test.ts

```diff
@@ -1,64 +1,364 @@
+        'use("first"); use("second"); db.collection.find({ t});',
+        { line: 0, character: 51 }
+      );
+
+      const jsCompletion = result.find(
+        (itme: { label: string; kind: number }) => itme.label === 'JavaScript'
+      );
+      const tsCompletion = result.find(
+        (itme: { label: string; kind: number }) => itme.label === 'TypeScript'
+      );
+
+      expect(jsCompletion).to.be.undefined;
+      expect(tsCompletion).to.have.property('kind', 1);
+    });
+
+    test('provide fields completion if function scope', async () => {
+      testMongoDBService.updatedCurrentSessionFields({
+        'test.collection': [
+          {
+            label: 'JavaScript',
+            kind: 1
+          }
+        ]
+      });
+
+      const result = await testMongoDBService.provideCompletionItems(
+        'use("test"); const name = () => { db.collection.find({ j}); }',
+        { line: 0, character: 56 }
+      );
+
+      const findCompletion = result.find(
+        (itme: { label: string; kind: number }) => itme.label === 'JavaScript'
+      );
 
-    await testMongoDBService.connectToServiceProvider({
-      connectionString: 'mongodb://localhost:27018',
-      connectionOptions: {}
+      expect(findCompletion).to.have.property('kind', 1);
     });
 
-    expect(testMongoDBService.connectionString).to.be.equal(
-      'mongodb://localhost:27018'
-    );
-    expect(testMongoDBService.connectionOptions).to.be.deep.equal({});
+    test('do not provide fields completion if has not db', async () => {
+      testMongoDBService.updatedCurrentSessionFields({
+        'test.collection': [
+          {
+            label: 'JavaScript',
+            kind: 1
+          }
+        ]
+      });
 
-    await testMongoDBService.disconnectFromServiceProvider();
+      const result = await testMongoDBService.provideCompletionItems(
+        'db.collection.find({ j});',
+        { line: 0, character: 22 }
+      );
+      const findCompletion = result.find(
+        (itme: { label: string; kind: number }) => itme.label === 'JavaScript'
+      );
 
-    expect(testMongoDBService.connectionString).to.be.undefined;
-    expect(testMongoDBService.connectionOptions).to.be.undefined;
+      expect(findCompletion).to.be.undefined;
+    });
+
+    test('do not provide fields completion if has wrong db', async () => {
+      testMongoDBService.updatedCurrentSessionFields({
+        'test.collection': [
+          {
+            label: 'JavaScript',
+            kind: 1
+          }
+        ]
+      });
+
+      const result = await testMongoDBService.provideCompletionItems(
+        'use("other"); db.collection.find({ j});',
+        { line: 0, character: 36 }
+      );
+      const findCompletion = result.find(
+        (itme: { label: string; kind: number }) => itme.label === 'JavaScript'
+      );
+
+      expect(findCompletion).to.be.undefined;
+    });
+
+    test('do not provide fields completion if has wrong collection', async () => {
```

Comment:
Should the provideCompletionItems be returning anything here? Maybe we can add not returning anything that as another `expect`.